src/Haskoin/Transaction/Partial.hs

```diff
@@ -679,15 +679,15 @@ instance Serialize PSBTHDPath where
         PSBTHDPath <$>
             S.isolate
                 (fromIntegral valueSize)
-                ((,) <$> S.getWord32le <*> getKeyIndexList numIndices)
+                ((,) <$> S.getWord32be <*> getKeyIndexList numIndices)
       where
         getKeyIndexList n = replicateM n S.getWord32le
 
     put (PSBTHDPath (fp, kis)) = do
         putVarInt (B.length bs)
         S.putByteString bs
       where
-        bs = S.runPut $ S.putWord32le fp >> mapM_ S.putWord32le kis
+        bs = S.runPut $ S.putWord32be fp >> mapM_ S.putWord32le kis
```

Comment:
What about the other word32le there?

Is there mixed byteorder?

Also, I'm surprised the test suite doesn't catch this!!

Would you mind checking if the psbt test suite is missing vectors?

Reply:
We might also need to change that.  `DerivPathI` uses big-endian byte order for key indices.  I'll add test vectors as needed and make sure that what we do conforms with the BIP 174 spec.  The BIP has a lot of test vectors.

Reply:
The situation is messy:
- BIP 32 specifies big-endian for key indices
- BIP 174 specifies little-endian for HD path components

Also, I added a test case covering HD paths.